source/crud/update.txt

```diff
@@ -0,0 +1,153 @@
+
+.. code-block:: javascript
+
+   {
+     <update operator>: { <field1>: <value1>, ... },
+     <update operator>: { <field2>: <value2>, ... },
+     ...
+   }
+
+Some update operators, such as :update:`$set`, create the field if
+the field does not exist. See the individual
+:manual:`update operator </reference/operator/update>` reference for
+details.
+
+Update a Single Document
+------------------------
+
+The following example uses the :method:`db.collection.updateOne()`
+method on the ``sample_mflix.movies`` collection to update the *first*
+document where ``title`` equals ``"Titanic"``:
+
+.. code-block::
+
+   use sample_mflix
+
+   db.movies.updateOne( { title: "Titanic" },
+   {
+     $set: { countries: [ "USA", "Canada" ] },
+     $currentDate: { lastUpdated: true }
+   }
+
+The update operation:
+
+- uses the :update:`$set` operator to update the value of the
+  ``countries`` array to ``[ "USA", "Canada" ]``,
+
+- uses the :update:`$currentDate` operator to update the value
+  of the ``lastUpdated`` field to the current date. If
+  ``lastUpdated`` field does not exist,
+  :update:`$currentDate` will create the field. See
+  :update:`$currentDate` for details.
+
+Update Multiple Documents
+-------------------------
+
+The following example uses the :method:`db.collection.updateMany()`
+method on the ``sample_airbnb.listingsAndReviews`` collection to update
+all documents where ``security_deposit`` is less than ``100``:
+
+.. code-block:: javascript
+
+   use sample_airbnb
+
+   db.listingsAndReviews.updateMany(
+     { security_deposit: { $lt: 100 } },
+     {
+       $set: { security_deposit: 100, minimum_nights: 1 }
+     }
+   )
+
+The update operation uses the :update:`$set` operator to update the
+value of the ``security_deposit`` field to ``100`` and the value of the
+``minimum_nights`` field to ``1``.
+
+Replace a Document
+------------------
+
+To replace the entire content of a document except for the ``_id``
+field, pass an entirely new document as the second argument to
+:method:`db.collection.replaceOne()`.
+
+When replacing a document, the replacement document must contain only
+field/value pairs. Do not include :manual:`update operators
+</reference/operator/update>` expressions.
+
+The replacement document can have different fields from the original
+document. In the replacement document, you can omit the ``_id`` field
+since the ``_id`` field is immutable; however, if you do include the
+``_id`` field, it must have the same value as the current value.
+
+The following example replaces the *first* document from the
+``sample_analytics.accounts`` collection where ``account_id: 371138``:
+
+.. code-block:: javascript
+
+   db.accounts.replaceOne(
+     { account_id: 371138 },
+     { account_id: 893421, limit: 5000, products: [ "Investment", "Brokerage" ] }
+   )
+
+Run the following command to read your updated document:
+
+.. code-block:: javascript
+
+   db.accounts.findOne( { account_id: 371138 } )
```

Comment:
note for review: this won't work because we updated the account id. Will fix in patch 2.